slapo/schedule.py

```diff
@@ -1093,6 +1142,18 @@ def init_target_engine(sch, target, **kwargs):
     )
 
 
+def reduce_model_grads(sch: Schedule):
```

Comment:
I like this idea, but not sure if there's any issue with this implementation. Maybe we could revisit it later.